srml/balances/src/lib.rs

```diff
@@ -287,7 +287,10 @@ impl<T: Trait> Module<T> {
 		let to_balance = Self::free_balance(&dest);
 		let would_create = to_balance.is_zero();
 		let fee = if would_create { Self::creation_fee() } else { Self::transfer_fee() };
-		let liability = value + fee;
+		let liability = match value.checked_add(&fee) {
```

Comment:
`let liability = ensure!(value.checked_add(&fee), "got overflow after adding a fee to value");`

Reply:
```rust
error[E0600]: cannot apply unary operator `!` to type `std::option::Option<<T as Trait>::Balance>`
   --> srml/balances/src/lib.rs:290:19
    |
290 |         let liability = ensure!(value.checked_add(&fee), "got overflow after adding a fee to value");
    |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ cannot apply unary operator `!`
    |
    = note: an implementation of `std::ops::Not` might be missing for `std::option::Option<<T as Trait>::Balance>`
    = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)
```

seems like `ensure!` works like an `assert!`, i.e. takes a `bool`

Expand shows me this

```rust
        let liability = {
            if !value.checked_add(&fee) {
                {
                    return Err("got overflow after adding a fee to value");
                };
            }
        };
```

Reply:
ahh yeah. fair enough.